recipe/meta.yaml

```diff
@@ -241,7 +1011,10 @@ outputs:
         - cmake
         - ninja
       host:
-        - {{ pin_subpackage('libarrow', exact=True) }}
+        - {{ pin_subpackage('libarrow-dataset', exact=True) }}
+        - {{ pin_subpackage('libarrow-flight', exact=True) }}
+        - {{ pin_subpackage('libarrow-gandiva', exact=True) }}
+        - {{ pin_subpackage('libarrow-substrait', exact=True) }}
```

Comment:
I guess there's a question hiding in there whether we want to provide a `libarrow-all` or something like that, which is just a metapackage that pulls together all the `lib*` components, and then `pyarrow` would host-depend on that.